### PR TITLE
[SYCLomatic] Fix segmentation fault when a template arg is a template type

### PR DESCRIPTION
**Pull request by AndyCHHuang**
Signed-off-by: Huang, Andy <andy.huang@intel.com>

### CODE REVIEW

**Review by ziranzha — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ExprAnalysis.cpp [hunk 1]
```diff
@@ -887,7 +906,12 @@ void ExprAnalysis::analyzeType(TypeLoc TL, const Expr *CSCE) {
       auto Result = Rewriter->rewrite();
       if (Result.hasValue()) {
         auto ResultStr = Result.getValue();
-        addReplacement(SR.getBegin(), SR.getEnd(), CSCE, ResultStr);
+        printf("SR %s\n", SR.getEnd().printToString(SM).c_str());
```
Comment (zhiweij1) [inline clang/lib/DPCT/ExprAnalysis.cpp]: please remove debug code

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**